python/astro_metadata_translator/translators/decam.py

```diff
@@ -393,6 +393,35 @@ def fix_header(
             modified = True
             log.debug("%s: Set FILTER to %s because OBSTYPE is %s", log_label, header["FILTER"], obstype)
 
+        # The following provides corrections to the boresight header info for
+        # the raw data files obtained from http://astroarchive.noirlab.edu/,
+        # program number 2013A-0351, of the Trifid and Lagoon Nebulae region.
+        if "Trifid" in header.get("OBJECT", "unknown") and obstype == "object":
+            date_obs = astropy.time.Time(header["DATE-OBS"], format="isot", scale="utc")
+            date_min = astropy.time.Time("2013-02-10", format="isot", scale="utc")
+            date_max = astropy.time.Time("2013-02-14", format="isot", scale="utc")
+            if date_obs > date_min and date_obs < date_max:
+                delta_ra = -0.027417  # median diff between header RA & https://astroarchive.noirlab.edu RA
+                delta_dec = -0.002833  # median diff between header Dec & https://astroarchive.noirlab.edu DEC
+                log.debug("Found Trifid in OBJECT header.  Shifting TELRA and TELDEC...")
+                radec_target = SkyCoord(
+                    header["RA"],
+                    header["DEC"],
+                    frame=header["RADESYS"].strip().lower(),
+                    unit=(u.hourangle, u.deg),
+                )
+                radec_fixed = SkyCoord(
+                    radec_target.ra.deg + delta_ra,
+                    radec_target.dec.deg + delta_dec,
+                    frame=header["RADESYS"].strip().lower(),
+                    unit=(u.deg, u.deg),
+                )
+                log.debug("Orig TELRA = {} TELDEC = {}".format(header["TELRA"], header["TELDEC"]))
+                header["TELRA"] = radec_fixed.ra.hourangle
```

Comment:
Why isn't this `header["TELRA"] += delta_ra` ? (assuming delta_ra is in hours) And `header["TELDEC"] += delta_dec`? Is all the SkyCoord calculating a distraction?

Reply:
Two issues.  First, it's actually:
```
header["TELRA"] = header["RA"] + delta_ra
header["TELDEC"] = header["DEC"] + delta_dec
```
(perhaps there was some convoluted mixup between the target vs. telescope header settings?)

Second, `header["TELRA"]` & `header["TELDEC"]` return `HH:MM:SS` & `DD:MM:SS` strings, respectively, e.g.
```
RA      = '18:02:33.333'       / [HH:MM:SS] Target RA                           
DEC     = '-23:46:59.999'      / [DD:MM:SS] Target DEC
```
So I need to convert the strings to degrees somehow.  I agree that there are other ways to do this, but I was following what is done in the `tracking_from_degree_headers()` function: https://github.com/lsst/astro_metadata_translator/blob/main/python/astro_metadata_translator/translators/helpers.py#L131-L139

If you think I should instead do the conversion outside of `SkyCorrd` (is doing the string-parsing back and forth considered clearer/simpler outside of `SkyCorrd`...perhaps I'm missing something?), suggestions welcome!

Reply:
Alllrighty, see if this `Angle`-only version looks better to you (I didn't realize the extra overhead of `SkyCoord` was worth worrying about!)  I've checked that it produces the same results and the relevant data process as expected.

And thanks for your patience and advice working me through this!